# Análisis de Datos Olímpicos: Puntuaciones y Proyecciones Matemáticas

Durante los Juegos Olímpicos de París 2024, se registraron las puntuaciones máximas de los equipos de Estados Unidos, China y Francia en tres disciplinas deportivas: gimnasia artística, saltos ornamentales (clavados) y tiro con arco.

El evento deportivo tuvo una duración de tres semanas, y se registraron las puntuaciones en las dos primeras semanas. Las matrices $A$ y $B$ representan las puntuaciones máximas de las semanas $1$ y $2$, respectivamente:

<center>
<img src="https://i.imgur.com/zGLb0un.png" style="width:50%">
</center>


Donde los elementos $a_{ij}$ y $b_{ij}$ representan los puntos obtenidos por el país $i$ en la disciplina deportiva $j$, siguiendo el mismo orden en que se mencionaron en el párrafo anterior.



Utiliza Python para responder las siguientes preguntas:

1. Determina si los países mejoraron su desempeño en cada disciplina durante la segunda semana con respecto a la primera semana. Muestra los resultados en una matriz $D$ e interpreta el elemento $d_{31}$.
1. Después de semanas de competencia la motivación de los deportistas se encuentra en su punto más alto. Debido a esto, analistas deportivos estimaron que en la tercera semana los deportistas podrían mejorar su rendimiento en torno a un $3\%$ respecto a la primera semana. Representa los resultados que se proyectan para la última semana en la matriz $E$  e interpreta el elemento $e_{22}$. Redondea los resultados a la centésima.
1. Uno de los criterios para determinar el país ganador en las Olimpiadas consiste en asignar puntos a cada tipo de medalla: $3$ puntos por una medalla de oro, $2$ puntos por una medalla de plata y $1$ punto por una medalla de bronce. Luego, se suman los puntos totales para determinar al país con mayor puntaje. Considera que los elementos $f_{ij}$ de la matriz $F$, según el orden en que se mencionan, representan el total de medallas de cada tipo $j$ (oro, plata y bronce) obtenidas por el país $i$ (Estados Unidos, China y Francia). Por otro lado, la matriz $G$ representa el puntaje asignado según el tipo de medalla.

<center>
<img src="https://i.imgur.com/uMMmSHH.png" style="width:31%">
</center>

$\hspace{0.5 cm}$ Representa mediante una matriz $H$ el puntaje total obtenido por cada país e interpreta la diferencia $h_{11} - h_{21}$.

4. Durante esta edición de los Juegos Olímpicos, un país obtuvo un total de $10$ medallas entre oro, plata y bronce, logrando un puntaje total de $17$ puntos ($3$ puntos por cada medalla de oro, $2$ puntos por cada medalla de plata y $1$ punto por cada medalla de bronce). Además, los deportistas de este país recibieron un aporte estatal de $53$ millones de pesos por cada medalla de oro, $40$ millones de pesos por cada medalla de plata y $20$ millones de pesos por cada medalla de bronce. Con estos datos, se plantea el siguiente sistema de ecuaciones:


<center>
<img src="https://i.imgur.com/8eCmpqQ.png" style="width:24%">
</center>

$\hspace{0.5 cm}$ Donde $x$, $y$ y $z$ representan la cantidad de medallas de oro, plata y bronce obtenidas por el país. Resuelve el sistema de ecuaciones utilizando matrices e interpreta los resultados obtenidos en el contexto del problema.

---

## Soluciones

#### Contexto

In [17]:
import numpy as np

- País: fila $i$
- Disciplina: col $j$

In [22]:
# Cols: Gimnasia artística, clavados y tiro
matrixA_sem1 = np.array([
    [9.5, 9.4, 9.0],  # USA
    [9.5, 9.6, 8.8],  # China
    [8.8, 9.3, 9.2],  # Francia
])
matrixB_sem2 = np.array([
    [9.6, 9.4, 9.2],  # USA
    [9.7, 9.7, 9.0],  # China
    [8.8, 9.1, 9.6],  # Francia
])

#### Respuesta 1.

In [23]:
matrixD_diferencia = matrixA_sem1 - matrixB_sem2
print(matrixD_diferencia)

[[-0.1  0.  -0.2]
 [-0.2 -0.1 -0.2]
 [ 0.   0.2 -0.4]]


##### USA

| Disciplina | Desempeño |
|---|---|
| Gimnasia Artística | Empeoró ligeramente su desempeño en $-0.1$ puntos |
| Clavados | Mantuvo su desempeño |
| Tiro con Arco | Empeoró su desempeño en $-0.2$ puntos |


##### China

| Disciplina | Desempeño |
|---|---|
| Gimnasia Artística | Empeoró su desempeño en $-0.2$ puntos |
| Clavados | Empeoró su desempeño en $-0.1$ puntos |
| Tiro con Arco | Empeoró su desempeño en $-0.2$ puntos |


##### Francia

| Disciplina | Desempeño |
|---|---|
| Gimnasia Artística | Mantuvo su desempeño |
| Clavados | Mejoró su desempeño en $0.2$ puntos |
| Tiro con Arco | Empeoró su desempeño en $-0.4$ puntos |

#### Respuesta 2.

In [24]:
mejora_rendimiento_esperada = 0.03
matrixE_proyeccion = mejora_rendimiento_esperada * matrixA_sem1

np.set_printoptions(precision=2)
print(matrixE_proyeccion)

[[0.28 0.28 0.27]
 [0.28 0.29 0.26]
 [0.26 0.28 0.28]]


In [25]:
print(f"{(matrixE_proyeccion[1,1] + matrixA_sem1[1,1]):.02f}")

9.89


El elemento $e_{22}$ ($0.29$) corresponde al **incremento** de **puntaje** con respecto a la primera semana de **China** en **clavados**. Es decir, se espera un puntaje de $9.89$.

#### Respuesta 3.

In [53]:
# Cols: Medallas de oro, plata, bronce
matrixF_suma_medallas = np.array([
    [42, 43, 31],  # USA
    [40, 35, 20],  # China
    [30, 15, 16],  # Francia
])

matrixG_puntaje_medalla = np.array([
    [3],  # Oro
    [2],  # Plata
    [1],  # Bronce
])

In [54]:
matrixH_puntajes = np.dot(matrixF_suma_medallas, matrixG_puntaje_medalla)

print("Matriz H:\n", matrixH_puntajes)

Matriz H:
 [[243]
 [210]
 [136]]


La diferencia $h_{11}-h_{21}$ representa la _diferencia de puntaje_ entre **USA** y **China**.

#### Respuesta 4.

In [55]:
A = np.array([
    [1, 1, 1],
    [3, 2, 1],
    [53, 40, 20],
])
B = np.array([
    [10],
    [17],
    [326],
])

A_inv = np.linalg.inv(A)
X = np.dot(A_inv, B)

print(X)

[[2.]
 [3.]
 [5.]]


In [58]:
# Revisamos:
assert 326 == (53 * 2) + (40 * 3) + (20 * 5)

Resolviendo obtenemos:

- $x=2$
- $y=3$
- $z=5$

Donde $x$ es la cantidad de medallas de oro, $y$ es la cantidad de medallas de plata y $z$ la cantidad de medallas de bronce.